### PR TITLE
New cvar amx_perfmon

### PR DESCRIPTION
**Pull request by UnrealKaraulov**
New cvar amx_perfmon

Now possible to disable amx_permon, or enable for it all plugins.

1. Disable
2. Enable for debug plugins
3. Enable for all plugins

### CODE REVIEW

**Comment by UnrealKaraulov** [comment]
@dvander you still alive? 👽 

Possible to merge this pr? 
It can add possibly use performance monitoring for plugins in release mode, more accurate than in debug.

**Comment by dvander** [comment]
Is debug mode required for perf monitor? Eg, should amx_debug be a bitmask, or a level?

**Comment by UnrealKaraulov** [comment]
> Is debug mode required for perf monitor? Eg, should amx_debug be a bitmask, or a level?
Perf monitor at this time works only for plugins what in debug mode. But this commit add possibly to use it for all plugins, even in release mode.

At this time in master, amx_debug can be 1, and 2.
0 = nothing (as 1)
1 = put in debug only plugins marked 'debug' in plugins.ini
2 = put all plugins in debug (and performance monitor)
3 = nothing (as 1)

This commit add new amx_debug mode '3'
3 = put in debug only plugins marked 'debug' in plugins.ini and force to use performance monitor for all plugins, even if not marked as debug.

**Comment by dvander** [comment]
It seems like it'd be better to keep amx_debug as-is, and then have a separate flag for the performance monitor.

**Comment by UnrealKaraulov** [comment]
> It seems like it'd be better to keep amx_debug as-is, and then have a separate flag for the performance monitor.

Something like this : https://github.com/alliedmodders/amxmodx/pull/1028/commits/376cfa0fbeddff25f2cdc0e1d53c320f525bafd0

?????

**Comment by UnrealKaraulov** [comment]
Updated.
> It seems like it'd be better to keep amx_debug as-is, and then have a separate flag for the performance monitor.

Something like this : https://github.com/alliedmodders/amxmodx/pull/1028/files ??

@dvander Now possible to disable amx_perf, or enable for all plugins. 

0. Disable
1. Enable for debug plugins
2. Enable for all plugins

**Comment by UnrealKaraulov** [comment]
@dvander now all okay or need another changes?

**Comment by UnrealKaraulov** [comment]
@dvander This is for previous PR. Now performance monitoring can be Disabled/Enabled for debug plugins/Enabled for anything. You can merge this PR? (because monitor can't be disabled without this fix)

**Comment by UnrealKaraulov** [comment]
I can compile it in another branch in GitHub actions with artifacts upload. But better if this will be merged. Possible all authors of amxx is dead ;(

**Comment by UnrealKaraulov** [comment]
@stambeto2006 you can download this version here https://github.com/UnrealKaraulov/amxmodx/actions/runs/1982492726

File: configs/amxx.cfg [hunk 1]
```diff
@@ -184,10 +184,20 @@ amx_mldebug ""
...
+// If function executed more than amx_perfmon_ms milliseconds
 // print plugin and function name to current amxmodx log file.
 //
 // Time in milliseconds
-// Default value: 1.0
+// Default value: 5.0
 // 
-amx_perflog_ms 1.0
\ No newline at end of file
+amx_perfmon_ms 5.0
```
Comment (wopox1337) [inline configs/amxx.cfg:203]:
```suggestion
amx_perfmon_ms 5.0

```

File: configs/cstrike/amxx.cfg [hunk 2]
```diff
@@ -223,10 +223,20 @@ amx_restrmapsettings 0
...
+// If function executed more than amx_perfmon_ms milliseconds
 // print plugin and function name to current amxmodx log file.
 //
 // Time in milliseconds
-// Default value: 1.0
+// Default value: 5.0
 // 
-amx_perflog_ms 1.0
\ No newline at end of file
+amx_perfmon_ms 5.0
```
Comment (wopox1337) [inline configs/cstrike/amxx.cfg:242]:
```suggestion
amx_perfmon_ms 5.0

```

File: configs/ns/amxx.cfg [hunk 3]
```diff
@@ -222,10 +222,20 @@ amx_unstuck_frequency 4
...
+// If function executed more than amx_perfmon_ms milliseconds
 // print plugin and function name to current amxmodx log file.
 //
 // Time in milliseconds
-// Default value: 1.0
+// Default value: 5.0
 // 
-amx_perflog_ms 1.0
\ No newline at end of file
+amx_perfmon_ms 5.0
```
Comment (wopox1337) [inline configs/ns/amxx.cfg:241]:
```suggestion
amx_perfmon_ms 5.0

```

**Review by wopox1337 — CHANGES_REQUESTED** [review]

**Comment by rtxa** [comment]
Hi @dvander,

Sorry for tagging you. Would you mind taking a look to this PR? The new *performance monitor* feature is good, but unfortunately it can't be disabled. Seems to me that @UnrealKaraulov made a very simple solution for this issue, where you can disabled it forever, enabled it only for plugins in debug mode or enabled it for all plugins.

Otherwise, it would obsfucated important information for developers not looking for this kind of info, or server owners where plugins are set in debug mode for other type of errors.

A workaround for this issue would be to set `amx_perflog_ms` to some high number like `9999` and it would disable the performance monitor.

**Comment by UnrealKaraulov** [comment]
@dvander @Arkshine америкосы

**Pull request closed without merging**